fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1129,6 +1122,28 @@ def forward(self, *args: Any, **kwargs: Any) -> torch.Tensor:
 
         return outputs
 
+    def _checkpoint_module_last_backward_call(self) -> bool:
+        # If this is a checkpointed FSDP module, e.g. checkpoint(FSDP(module)),
+        # we check if the following counter reaches 0. If it is, it is the last
+        # inner backward call for this FSDP module.
+        if (
+            hasattr(self._fsdp_wrapped_module, "_checkpoint_fwd_counter")
+            and self._fsdp_wrapped_module._checkpoint_fwd_counter != 0
+        ):
+            return False
+        return True
+
+    def _require_final_backward(self) -> bool:
```

Comment:
`@property` ?

Reply:
I'm not sure whether it should be @property, as it is only called by root